src/pkg/packager/validator/validator.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package validator contains functions for verifying zarf yaml files are valid
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	validatorInvalidPrefix = "schema is invalid:"
+	validatorWarningPrefix = "zarf schema warning:"
+)
+
+// Validator holds the warnings/errors and messaging that we get from validation
+type Validator struct {
+	warnings           []string
+	errors             []error
+	jsonSchema         []byte
+	typedZarfPackage   types.ZarfPackage
+	untypedZarfPackage interface{}
+}
+
+func (v Validator) getFormatedError() error {
+	if !v.hasErrors() {
+		return nil
+	}
+	errorMessage := validatorInvalidPrefix
+	for _, errorStr := range v.errors {
+		errorMessage = fmt.Sprintf("%s\n - %s", errorMessage, errorStr.Error())
+	}
+	return errors.New(errorMessage)
+}
```

Comment:
Should we implement the `Error` interface here? https://gobyexample.com/errors

Reply:
I'm leaning towards that we shouldn't implement error since I feel it would imply that the validator object is an error while it isn't really. I could see it both ways though, curious what you think @Racer159

Reply:
I had switched validator to implement the error interface, but I ended up deleting this function all together to instead print a table.